### PR TITLE
[CCM-11324]: first draft of the docs

### PR DESCRIPTION
**Pull request by joeyouss**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
+# Introduction to Commitment Orchestrator
+In recent years, organizations leveraging cloud services have witnessed a concerning trend - the steady rise in cloud costs. As businesses increasingly migrate their operations to the cloud, the expenses associated with cloud infrastructure, storage, and data processing have become a significant portion of the cloud bills..
+
+To help bring down the cloud costs, providers often offer special discounted rates against on-demand costs to the customers. These commitments are mostly in form of **RI (Reserved Instances)** or **SPs(Savings Plan)**. 
+
+Reserved Instances are a pricing model offered by cloud service providers that allows customers to reserve and commit to a specific amount of computing capacity in advance for a term of one or three years. This commitment comes with a significant discount compared to on-demand pricing, making it a cost-effective option for workloads with predictable and sustained usage.
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:16]: @joeyouss Please add the definition of SP as well and also mention what they are called in each cloud provider's context.

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+However, it's essential for organizations to carefully analyze their usage patterns before making commitments to ensure they choose the most suitable Reserved Instances for their needs.
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:18]: Everywhere please add Saving plan along with RIs. @joeyouss

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+description: This topic describes Commitment Orchestrator and its working.
+# sidebar_position: 2
+helpdocs_topic_id: 
+helpdocs_category_id: 
+helpdocs_is_private: false
+helpdocs_is_published: true
+---
+
+# Introduction to Commitment Orchestrator
+In recent years, organizations leveraging cloud services have witnessed a concerning trend - the steady rise in cloud costs. As businesses increasingly migrate their operations to the cloud, the expenses associated with cloud infrastructure, storage, and data processing have become a significant portion of the cloud bills..
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:12]: We should not call it as a concerning trend but rather a growing business implying to huge cloud costs as most organisations are either moving from on-prem or increasing their business in cloud front.

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+However, it's essential for organizations to carefully analyze their usage patterns before making commitments to ensure they choose the most suitable Reserved Instances for their needs.
+
+Commitments in the cloud, such as Reserved Instances (RIs) or other long-term contracts, can pose challenges in various situations if not managed correctly.
+
+If the organization's workload patterns or compute spend changes significantly due to policy or market factors, commitments made based on initial assumptions may no longer align with actual resource needs.Rapid changes in technology or the adoption of new services may render existing commitments less cost-effective or even obsolete.
+Overcommitting to a specific cloud provider or service may lead to vendor lock-in, making it challenging to switch to a different provider or adopt a multi-cloud strategy. Economic or business uncertainties can make long-term commitments risky, especially if there's a chance of downsizing or changes in strategy.
+
+Some commitments, such as rigid long-term contracts, may lack flexibility to adapt to changing circumstances or unexpected events
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:25]: What is referred to as rigid long term contracts here?

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+Some commitments, such as rigid long-term contracts, may lack flexibility to adapt to changing circumstances or unexpected events 
+Overcommitting resources may result in underutilization, leading to inefficient use of commitments and potentially higher overall costs due to unnecessary commitment plans.
+Overhead in management of commitments: While managing commitments, there's always overhead in understanding the context of the need of purchase and to identify what to purchase next among the multiple options for commitments, It requires continuous manual effort and is a lot more human error prone in case of wrong predictions.
+
+At Harness, we are committed to bring in the power of **AI and ML** to manage the commitments. Harness CCM helps you to track everything about your cloud spends so that you have an optimized cloud spend. With our latest addition of Commitment Orchestrator, we focus on managing your commitments for optimal performance and mitigating the problems associated with automated management of commitments.
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:29]: "mitigating the problems associated with automated management of commitments."  This should be manual management of commitments not automated. @joeyouss

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+## What is over commitment and under commitment?
+In the context of cloud spend or cloud bills breakdown,, over-commitment and under-commitment typically refer to the utilization of resources compared to the commitments made through mechanisms like Reserved Instances (RIs) or Savings Plan (SPs).
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:32]: commitments made through discount plans (not mechanism)

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+At Harness, we are committed to bring in the power of **AI and ML** to manage the commitments. Harness CCM helps you to track everything about your cloud spends so that you have an optimized cloud spend. With our latest addition of Commitment Orchestrator, we focus on managing your commitments for optimal performance and mitigating the problems associated with automated management of commitments.
+
+## What is over commitment and under commitment?
+In the context of cloud spend or cloud bills breakdown,, over-commitment and under-commitment typically refer to the utilization of resources compared to the commitments made through mechanisms like Reserved Instances (RIs) or Savings Plan (SPs). 
+
+### Over-Commitment:
+Over-commitment occurs when an organization has reserved more resources (such as instances or capacity) or bought extra Savings Plans to cover their compute spend than it actually needs or utilizes.
+For example, an organization may have purchased a significant number of Reserved Instances based on an initial estimation of their workload. If the actual usage turns out to be lower than expected, the organization is considered to be over-committed.
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:36]: @joeyouss  orgnaisation is not considered to be over-committed, but the resources they have deployed are over committed leading to wastage and under utlisaition .

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+## How does Commitment Orchestrator do this?
+Commitment Orchestrator does all of this with the help of its 2 engines: **Utilisation engine and Purchase engine**
+
+| **Utilisation engine** | **Purchase engine** |
+| --- | --- |
+| The utilization engine leverages machine learning (ML) and intelligent analysis of existing usage patterns to optimize resource utilization and increase the efficiency of underutilized Reserved Instances (RIs). It breaks RIs into the smallest units and assembles them to meet target instance types. By intelligently analyzing these patterns, the engine identifies opportunities for optimization that can lead to significant cost savings for the business. | The purchase engine analyzes  the on-demand spend and then purchases the RIs/SPs basis the target coverage.|
+
+
+## Getting Started
+Commitment Orchestrator can be found on our dashboard under Cloud Costs. Commitment Orchestrator can be easily setup from the dashboard itself by following these steps:
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:65]: @joeyouss under cloud costs meaning?

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+### Step 1 : Setting up the master account 
+
+You need to select the master account with the right permissions to be added via connector on which you want to enable orchestration. You can either select an existing connector for your master account or create one. 
+<DocImage path={require('./static/Step1.png')} width="50%" height="50%" title="Click to view full size image" />
+
+
+### Step 2: Exclude Instance types 
+Commitment Orchestrator provides you with an option to choose the instances you would like to exclude when orchestrating instances for Compute Usage.
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:73]: @joeyouss Note:  The purchases will happen only at master account level and thus will be in turn applicable for child accounts as well. The exclusion list will only be considered for the compute spend calulations and actual RI/SP may be used against the instances if they are part of child accounts.

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+<DocImage path={require('./static/Step2.png')} width="50%" height="50%" title="Click to view full size image" />
+
+
+### Step 3: Coverage Percentage and SP/RI split
+Coverage Percentage: The Coverage percentage is the percentage that you want to be covered by Savings Plans or Reserved Instances. The remaining amount will continue to run as on-demand.
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:78]: you want to cover your compute spend by SP/RIs. The remaining coverage will continue...

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+SP/RI split: Out of the totally compute spend, you have the option to split it between Convertible Reserved Instances and Compute Savings Plan
```
Comment (Additi) [inline docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:80]: Out of total coverage, the system automatically calculates the maximum guardrail for purchasing the Savings Plans given the past compute spend data to ensure limiting any over purchase of saving plans. You can reduce the percentage by increasing share of Reserved Instances. However, you can't increase the coverage via Savings Plan beyond the max limit.

**Review by Additi — CHANGES_REQUESTED** [review]
@joeyouss Please review the changes and ensure consistency on terms across in the whole page.

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md (hunk 1 above)
Comment (joeyouss) [reply docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:16]: done

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md (hunk 4 above)
Comment (joeyouss) [reply docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:25]: commitments made with cloud service providers for a fixed duration - by rigid I mean that once they sign contract, the organisation Is locked into using the agreed-upon resources

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md (hunk 5 above)
Comment (joeyouss) [reply docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:29]: Amm, in one comment on the doc, it was mentioned "The things that we do: 
Automated commitment purchasing and management"

so we do automated purchasing only?

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md (hunk 6 above)
Comment (joeyouss) [reply docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:32]: done

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md (hunk 7 above)
Comment (joeyouss) [reply docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:36]: done

File: docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md (hunk 8 above)
Comment (joeyouss) [reply docs/cloud-cost-management/4-use-ccm-cost-optimization/Commitment-orchestrator/overview.md:65]: https://app.harness.io/ - added backlink for this

**Review by Additi — APPROVED** [review]

**Pull request merged**